File: Engine.Core/SPECIAL/Experience.cs
using Engine.Core.Scripting;

namespace Engine.Core.SPECIAL
{
    public class Experience
    {
        private readonly ScriptRepository scriptRepository;

        public Experience()
        {
            scriptRepository = new ScriptRepository();
        }

        public const int MaxLevel = 30;

        public int Level { get; } = 1;
        public int CurrentExperience { get; } = 0;
        public object RemainingExperienceToNextLevel => scriptRepository.GetById("calculateExperienceNeededForNextLevel").Execute();

        public override string ToString() => $"{CurrentExperience}/{RemainingExperienceToNextLevel} {Level}/{MaxLevel}";
    }
}

File: src/app/model/entity/experience.ts
import {noLessThanZero} from "../../utils/utils";

export class Fallout2Experience {
  level = 1;
  currentExperience = 0;
  static readonly maxLevel = 99;

  calculateExperienceNeededForNextLevel = () => noLessThanZero(this.currentExperience - ((this.level * (this.level - 1) / 2) * 1000));
}
